integ/scripts/bash/rfdk-integ-e2e.sh

```diff
@@ -64,95 +69,51 @@ fi
 
 # Mark infrastructure deploy finish time
 export INFRASTRUCTURE_DEPLOY_FINISH_TIME=$SECONDS
 
-# Pull the top level directory for each cdk app in the components directory
-COMPONENTS=()
-for COMPONENT in **/cdk.json; do
-    # In case the yarn install was done inside this integ package, there are some example cdk.json files in the aws-cdk
-    # package we want to avoid.
-    if [[ $COMPONENT == *"node_modules"* ]]; then
-        continue
-    fi
-
-    COMPONENT_ROOT="$(dirname "$COMPONENT")"
-    COMPONENT_NAME=$(basename "$COMPONENT_ROOT")
-
-    # Use a pattern match to exclude the infrastructure app from the results
-    export ${COMPONENT_NAME}_START_TIME=$SECONDS
-    if [[ "$COMPONENT_NAME" != _* ]]; then
-        # Excecute the e2e test in the component's scripts directory
-        cd "$INTEG_ROOT/$COMPONENT_ROOT"
-        if [ "${RUN_TESTS_IN_PARALLEL-}" = true ]; then
-            ( (../common/scripts/bash/component_e2e.sh "$COMPONENT_NAME"; exit_code=$?; echo $exit_code > "$INTEG_TEMP_DIR/${COMPONENT_NAME}_exitcode"; exit $exit_code) \
-            || ../common/scripts/bash/component_e2e.sh "$COMPONENT_NAME" --destroy-only) &
-            export ${COMPONENT_NAME}_PID=$!
-            COMPONENTS+=(${COMPONENT_NAME})
-        else
-            ../common/scripts/bash/component_e2e.sh "$COMPONENT_NAME" || cleanup_on_failure
-        fi
-    fi
-    export ${COMPONENT_NAME}_FINISH_TIME=$SECONDS
-done
-
-if [ "${RUN_TESTS_IN_PARALLEL-}" = true ]; then
-    while [ "${#COMPONENTS[@]}" -ne 0 ]; do
-        ACTIVE_COMPONENTS=()
-        for COMPONENT_NAME in ${COMPONENTS[@]}; do
-            PID=$(eval echo \"\$${COMPONENT_NAME}_PID\")
-            if ps -p "$PID" > /dev/null; then
-              ACTIVE_COMPONENTS+=(${COMPONENT_NAME})
-            else
-              COMPONENT_EXIT_CODE=$(cat "$INTEG_TEMP_DIR/${COMPONENT_NAME}_exitcode" || echo 1)
-              if [ $COMPONENT_EXIT_CODE -ne 0 ]; then
-                SCRIPT_EXIT_CODE=1
-              fi
-
-              echo "Test app $COMPONENT_NAME finished with exit code $COMPONENT_EXIT_CODE"
-              if [ -f "$INTEG_TEMP_DIR/${COMPONENT_NAME}_deploy.txt" ]; then
-                cat "$INTEG_TEMP_DIR/${COMPONENT_NAME}_deploy.txt"
-              fi
-              if [ -f "$INTEG_TEMP_DIR/${COMPONENT_NAME}.txt" ]; then
-                cat "$INTEG_TEMP_DIR/${COMPONENT_NAME}.txt"
-              fi
-              if [ -f "$INTEG_TEMP_DIR/${COMPONENT_NAME}_destroy.txt" ]; then
-                cat "$INTEG_TEMP_DIR/${COMPONENT_NAME}_destroy.txt"
-              fi
-            fi
-            export ${COMPONENT_NAME}_FINISH_TIME=$SECONDS
-        done
-        if [ "${#ACTIVE_COMPONENTS[@]}" -ne 0 ]; then
-          COMPONENTS=(${ACTIVE_COMPONENTS[@]})
-        else
-          COMPONENTS=()
-        fi
-        sleep 1
-    done
-
-    wait
+XARGS_ARGS="-n 1"
+if [[ "${RUN_TESTS_IN_PARALLEL-}" = true ]]
+then
+  # Unbounded parallelism
+  XARGS_ARGS="${XARGS_ARGS} -P 0"
```

Comment:
Maybe expand on this comment to say that the `-P 0` flag will tell xargs to run all the commands in parallel and then block until they complete execution.